python/nano/src/bigdl/nano/pytorch/inference/optimizer.py

```diff
@@ -139,8 +139,15 @@ def optimize(self, model: nn.Module,
                if you would like to use the accelerated model in an online service.
         :param validation_data: (optional) A pytorch dataloader for accuracy evaluation
                This is only needed when users care about the possible accuracy drop.
-        :param metric: (optional) A callable object takes prediction and target
-               and returns a accuracy value in this calling method `metric(pred, target)`
+        :param metric: (optional) A callable object which is used for calculating accuracy.
+               1. If validation_data is composed of (input_data, target), metric could be
+               a torchmetrics.metric.Metric or similar object or function which takes
+               prediction and target then returns an accuracy value in this
+               calling method `accuracy = metric(pred, target)`.
+               2. Otherwise you can define a custom metric in this calling
+               method `accuracy = metric(model, batch_data)`.
+               3. Metric can even only recieve model as input in this calling
+               method `accuracy = metric(model)`, like coco_evaluator.
```

Comment:
The logic seems a little bit messy. Maybe it's simpler if we just require the metric be a callable object that takes model and data_loader (if not None) as input.

We can provide a helper class such as `NanoMetric` to help user compute the result if the user's metric takes (input_data, target), similar to what `_accuracy_calculate_helper` does; the user can then simply wrap their metric using `NanoMetric`

In addition, we can also automatically wrap the metric using `NanoMetric` inside `optimize` if it is a `TorchMetrics` object.

Reply:
Thanks for such great suggestions. I have modified code according to these comments.  Could you please take a look again to confirm that whether the current implementation is consistent with your vision ? @jason-dai 

Another question is, here I just ask user to provide a callable object that takes model and dataloader as input. Should I impose certain restrictions on this behavior, for example, given a Base class and ask user to inherit this class to implement custom logic?
```python
class BaseMetric(metaclass=ABCMeta):
    def __init__(self, metric: Callable):
        self.metric = metric
    
    @abstractmethod
    def __call__(self, model: nn.Module, data_loader: DataLoader = None):
        pass

class CustomMetric(BaseMetric):
    def __init__(self, metric: Callable):
        self.metric = metric
    
    def __call__(self, model: nn.Module, data_loader: DataLoader = None):
        # custom logic
        pass

metric = Accuracy()
metric = CustomMetric(metric)
```